plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/actions/DeleteAction.java

```diff
@@ -119,7 +121,31 @@ private boolean isValidSelection(IStructuredSelection selection) {
 			return false;
 		}
 
-		return true;
+		if ((this instanceof ActionAllowedInExternalSubmodel) || !hasExternalFeature(selection)) {
+			return true;
+		}
+
+		return false;
+
+	}
+
+	/**
+	 * @param selection
+	 * @return
+	 */
+	private boolean hasExternalFeature(IStructuredSelection selection) {
```

Comment:
Seems equivalent to the method in MultipleSelectionAction. It would be better to use the same method in this case.